src/jit/gentree.cpp

```diff
@@ -10166,6 +10166,18 @@ GenTreePtr                  Compiler::gtFoldExprConst(GenTreePtr tree)
                             goto LNG_OVF;
                         }
                         if ((ltemp/lval2) != lval1) goto LNG_OVF;
+#ifdef UNIX_AMD64_ABI
+                        // There is a clang 3.5 optimizer bug that in case of lval1 and lval2 equal
+                        // of 0x80000000 (MIN_LONG) the above expression (ltemp / lval2) results
+                        // in 0x80000000 (MIN_LONG. This causes an overflowing mul expression 
+                        // to be repalced with a 0, instead of throwing an overflow exception.
```

Comment:
I was curious and taken a look at the generated assembly code and the code seems correct to me. It loads the ltemp to RAX, then sign extends it to 128 bits in RDX:RAX and then divides it by lval2 and compares it to lval1.
What was the value of the ltemp in your case?
```asm
;10168                           if ((ltemp/lval2) != lval1) goto LNG_OVF;
   0x00007ffff652a349 <+9929>:  mov    -0x50(%rbp),%rax
   0x00007ffff652a34d <+9933>:  cqto
   0x00007ffff652a34f <+9935>:  idivq  -0x48(%rbp)
   0x00007ffff652a353 <+9939>:  cmp    -0x40(%rbp),%rax
```